File: frontend/packages/utils/src/images.ts
import { isProd } from "./env";

export function getGcsImageUrl(storageID: string) {
  const bucketName = isProd() ? "user-images-bucket-us" : "dev-user-images-bucket";
  // TODO: put images.trycoaster.com here for Prod to get CDN
  return `https://storage.googleapis.com/${bucketName}/${storageID}`;
}

File: backend/common/images/images.go
package images

import (
	"fmt"

	"go.coaster.io/server/common/application"
)

func GetGcsImageUrl(storageID string) string {
	var bucketName string
	if application.IsProd() {
		bucketName = "user-images-bucket-us"
	} else {
		bucketName = "dev-user-images-bucket"
	}

	// TODO: put images.trycoaster.com here for Prod to get CDN
	return fmt.Sprintf("https://storage.googleapis.com/%s/%s", bucketName, storageID)
}
